NetKet/Graph/hypercube.hpp

```diff
@@ -136,59 +125,93 @@ class Hypercube : public AbstractGraph {
       }
     }
 
-    symm_table_ = BuildSymmTable(L_, ndim_, pbc_, nsites_);
+    symm_table_ = BuildSymmTable(length_, n_dim_, pbc_, n_sites_);
   }
 
-  int Nsites() const override { return nsites_; }
+  int Nsites() const noexcept override { return n_sites_; }
 
-  int Size() const override { return nsites_; }
+  int Size() const noexcept override { return n_sites_; }
 
-  int Length() const { return L_; }
+  int Length() const noexcept { return length_; }
 
-  int Ndim() const { return ndim_; }
+  int Ndim() const noexcept { return n_dim_; }
 
   const std::vector<Edge> &Edges() const noexcept { return edges_; }
 
   std::vector<std::vector<int>> AdjacencyList() const override {
     return detail::AdjacencyListFromEdges(Edges(), Nsites());
   }
 
-  bool IsBipartite() const override { return !pbc_ || L_ % 2 == 0; }
+  bool IsBipartite() const noexcept override {
+    return !pbc_ || length_ % 2 == 0;
+  }
 
-  bool IsConnected() const override { return true; }
+  bool IsConnected() const noexcept override { return true; }
 
   // Returns map of the edge and its respective color
-  const ColorMap &EdgeColors() const override { return colors_; }
+  const ColorMap &EdgeColors() const noexcept override { return colors_; }
 
   std::vector<std::vector<int>> SymmetryTable() const override {
     return symm_table_;
   }
 
+  /// Given a site's coordinate as a `Ndim()`-dimensional vector, returns the
+  /// site's index. The mapping is unique, but unspecified. I.e. the fact that
+  /// sometimes `Coord2Site({1, 2, 0, 0, 1}) == X` and `Coord2Site({1, 2, 0, 0,
+  /// 2}) == X+1` should not be relied on.
+  int Coord2Site(std::vector<int> const &coord) const {
```

Comment:
We would also need the inverse mapping, from site to coordinates. This was previously returned by the member `Sites`